best_practices/controllers.rst

```diff
@@ -164,13 +177,14 @@ manually. In our application, we have this situation in ``CommentController``:
 
 .. code-block:: php
 
+    use Doctrine\ORM\EntityManagerInterface;
+
     /**
      * @Route("/comment/{postSlug}/new", name = "comment_new")
      */
-    public function newAction(Request $request, $postSlug)
+    public function newAction(Request $request, $postSlug, EntityManagerInterface $em)
```

Comment:
I don't have strong opinion on it, but if type hinting on the interface become the best practice, maybe should we consider to deprecate `getDoctrine` method in `4.x`? I'm not disturbed by the interface, it's only an interface you know it's not that terrible :)